### PR TITLE
Add stop rollup job support to HL REST Client

### PR DESCRIPTION
**Pull request by cbuescher**
This change adds support for stoping a rollup job to the High Level REST Client.

Relates to #29827

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@hub-cap I think this PR is largely analogous to #34623, would you mind taking a look?

**Comment by polyfractal** [comment]
Rollup stuff LGTM.  The client stuff does too, but I'll let @hub-cap cover that if he wants :)

Aside and unrelated to this PR: we should probably adjust `AcknowledgedResponse` to not use `ToXContent` in a later PR, since it is used by a number of other client APIs.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java [hunk 1]
```diff
@@ -55,8 +56,19 @@ static Request startJob(final StartRollupJobRequest startRollupJobRequest) throw
...
-        return request;
+        return new Request(HttpPost.METHOD_NAME, endpoint);
+    }
+
+
+    static Request stopJob(final StopRollupJobRequest stopRollupJobRequest) throws IOException {
+        String endpoint = new RequestConverters.EndpointBuilder()
+            .addPathPartAsIs("_xpack")
+            .addPathPartAsIs("rollup")
+            .addPathPartAsIs("job")
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java]: `.addPathPartAsIs("_xpack", "rollup", "job")`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java [hunk 2]
```diff
@@ -55,8 +56,19 @@ static Request startJob(final StartRollupJobRequest startRollupJobRequest) throw
...
+            .addPathPart(stopRollupJobRequest.getJobId())
+            .addPathPart("_stop")
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java]: `            .addPathPartAsIs("_stop")`

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RollupRequestConvertersTests.java [hunk 3]
```diff
@@ -61,6 +62,18 @@ public void testStartJob() throws IOException {
...
+    public void testStopJob() throws IOException {
+        String jobId = randomAlphaOfLength(5);
+
+        StopRollupJobRequest stopJob = new StopRollupJobRequest(jobId);
+
+        Request request = RollupRequestConverters.stopJob(stopJob);
+        assertThat(request.getEndpoint(), equalTo("/_xpack/rollup/job/" + jobId + "/_stop"));
+        assertThat(HttpPost.METHOD_NAME, equalTo(request.getMethod()));
+        assertThat(request.getParameters().keySet(), empty());
+        assertThat(request.getEntity(), nullValue());
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/RollupRequestConvertersTests.java]: `assertNull(request.getEntity())`

**Review by hub-cap — APPROVED** [review]
super minor nits

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java [hunk 4]
```diff
@@ -54,8 +55,19 @@ static Request startJob(final StartRollupJobRequest startRollupJobRequest) throw
             .addPathPart(startRollupJobRequest.getJobId())
             .addPathPartAsIs("_start")
             .build();
-        Request request = new Request(HttpPost.METHOD_NAME, endpoint);
-        return request;
+        return new Request(HttpPost.METHOD_NAME, endpoint);
+    }
+
+
```
[the remark this answers is not in the document]
Comment (cbuescher) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java]: gone

**Comment by cbuescher** [comment]
@hub-cap thanks for the review, I pushed another commit adressing your comments and am waiting on the green light from CI now.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Pull request merged**